### PR TITLE
add href attribute

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/HrefHandler.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+            protected override IEnumerable<PropInfo> GetProps(Type type)
+            {
+                return from prop in base.GetProps(type)
+                       where prop.Prop.GetSetMethod() != null
+                       where !prop.Prop.IsDefined(typeof(HrefIgnoreAttribute), false)
+                       select prop;
+
+            }
+            private string GetHrefFromRoot(string originalHref, HandleModelAttributesContext context)
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/HrefHandler.cs]: >private [](start = 12, length = 7)

minor: prepend a line

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/HrefHandler.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.DataContracts.Common;
+    using Microsoft.DocAsCode.Plugins;
+
+    public class HrefHandler : IModelAttributeHandler
+    {
+        private readonly ConcurrentDictionary<Type, HrefHandlerImpl> _cache = new ConcurrentDictionary<Type, HrefHandlerImpl>();
+
+        public object Handle(object obj, HandleModelAttributesContext context)
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/HrefHandler.cs]: also check context?

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/HrefHandler.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                    sources = new List<LinkSourceInfo>();
+                    context.FileLinkSources[file] = sources;
+                }
+                sources.Add(new LinkSourceInfo
+                {
+                    Target = file,
+                    Anchor = UriUtility.GetFragment(originalHref),
+                    SourceFile = ft.File,
+                });
+                context.LinkToFiles.Add(file);
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/HrefHandler.cs]: >LinkToFiles [](start = 24, length = 11)

when href is current file, should it be added into LinkToFiles?

File: test/Microsoft.DocAsCode.Build.Common.Tests/ModelAttributeHandlerTest.cs [hunk 4]
```diff
@@ -364,6 +391,8 @@ public class SimpleModel
             public string Identity { get; set; }
             [UniqueIdentityReference]
             public List<object> Identities { get; set; }
+            [Href]
```
Comment (vicancy) [inline test/Microsoft.DocAsCode.Build.Common.Tests/ModelAttributeHandlerTest.cs]: >Href [](start = 13, length = 4)

is the naming too general?...

File: test/Microsoft.DocAsCode.Build.Common.Tests/ModelAttributeHandlerTest.cs [hunk 5]
```diff
@@ -263,11 +263,34 @@ public void TesteModelWithIListMarkdownContentAttributeShouldSucceed()
...
         #endregion
 
+        #region HrefAttribute
+
+        [Fact]
+        public void TestSimpleModelWithHrefAttributeShouldSucceed()
+        {
+            var model = new SimpleModel
+            {
+                Href = "linkTarget"
```
Comment (vicancy) [inline test/Microsoft.DocAsCode.Build.Common.Tests/ModelAttributeHandlerTest.cs]: >Href [](start = 16, length = 4)

so before processing, it is the href relative to current model, and after processing, it is the href relative to root?

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/HrefHandler.cs (hunk 3 above)
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/HrefHandler.cs]: Seems it also makes sense to add

---
In reply to: [111533603](https://github.com/dotnet/docfx/pull/1543#discussion_r111533603) [](ancestors = 111533603)

File: test/Microsoft.DocAsCode.Build.Common.Tests/ModelAttributeHandlerTest.cs (hunk 5 above)
Comment (superyyrrzz) [inline test/Microsoft.DocAsCode.Build.Common.Tests/ModelAttributeHandlerTest.cs]: yes, like what Markup does.

---
In reply to: [111536280](https://github.com/dotnet/docfx/pull/1543#discussion_r111536280) [](ancestors = 111536280)

File: test/Microsoft.DocAsCode.Build.Common.Tests/ModelAttributeHandlerTest.cs (hunk 4 above)
Comment (superyyrrzz) [inline test/Microsoft.DocAsCode.Build.Common.Tests/ModelAttributeHandlerTest.cs]: yes... renamed to UrlContent

---
In reply to: [111536147](https://github.com/dotnet/docfx/pull/1543#discussion_r111536147) [](ancestors = 111536147)

**Comment by hellosnow** [comment]
:shipit:

**Comment by vicancy** [comment]
:shipit:

**Pull request merged**